modules/iap/iap.tf

```diff
@@ -0,0 +1,176 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# fetch all namespaces
+data "kubernetes_all_namespaces" "allns" {}
+
+resource "random_string" "random" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+# Jupyter IAP
+resource "google_compute_global_address" "jupyter_ip_address" {
+  count        = var.jupyter_add_auth && var.jupyter_url_domain_addr == "" ? 1 : 0
+  provider     = google-beta
+  project      = var.project_id
+  name         = "jupyter-address-${random_string.random.result}"
+  address_type = "EXTERNAL"
+  ip_version   = "IPV4"
+}
+
+# Helm Chart IAP
+resource "helm_release" "iap_jupyter" {
+  count            = var.jupyter_add_auth ? 1 : 0
```

Comment:
rename var jupyter_add_auth

Reply:
the same

Reply:
Is it possible to implement the specifics for Jupyter outside of the more generic IAP module?